web_tree_many2one_clickable/static/src/js/web_tree_many2one_clickable.js

```diff
@@ -53,14 +53,18 @@ ListView.Column.include({
             if (this.widget == 'many2one_unclickable') {
                 return name;
             } else if (this.use_many2one_clickable) {
+                let value = row_data[this.id].value;
                 var values = {
                     model: this.relation,
-                    id: row_data[this.id].value[0],
+                    id: value[0],
                     name: name,
                 };
-                if(this.type == 'reference' && !!row_data[this.id + '__display']) {
-                    values.model = row_data[this.id].value.split(',', 1)[0];
-                    values.id = row_data[this.id].value.split(',', 2)[1];
+                if(this.type == 'reference') {
+                    values.model = value.split(',', 1)[0];
+                    values.id = value.split(',', 2)[1];
+                    values.name = value;
+                }
+                if (!!row_data[this.id + '__display']) {
```

Comment:
Here perhaps can use a more readable syntax:
```suggestion
                if (_.has(row_data, this.id + '__display')) {
```